### ISSUE TITLE
Set up Mkdocs

### ISSUE THREAD

**Issue by adswa** [adswa/mpi-datamanagement-ws#1]
@lnnrtwttkhn mentioned Mkdocs, which sounds ideal for creating notes and material and rendering it. [Deploying via Github pages](https://www.mkdocs.org/user-guide/deploying-your-docs/) sounds easy. Continuous integration with Travis [should be doable](https://stackoverflow.com/questions/50892018/auto-build-an-mkdocs-documentation-in-travis-ci).
Could you set it up initially, @lnnrtwttkhn, and then we'll see to get it deployed and tested?

**Comment by lnnrtwttkhn** [comment]
I now setup basic documentation using [mkdocs](https://www.mkdocs.org/) in 69471a5 on branch `docs`. To serve the documentation locally you need to:
```bash
pip install mkdocs
cd mpi-datamanagement-ws
mkdocs serve
```
This will open up a local server on `http://127.0.0.1:8000` that one can view in a browser

**Comment by lnnrtwttkhn** [comment]
Deployment with GitHub pages seems indeed as easy as:
```bash
mkdocs gh-deploy
```
which results in 
```bash
mkdocs gh-deploy
INFO    -  Cleaning site directory 
INFO    -  Building documentation to directory: /Users/wittkuhn/edu/mpi-datamanagement-ws/site 
WARNING -  Version check skipped: No version specificed in previous deployment. 
INFO    -  Copying '/Users/wittkuhn/edu/mpi-datamanagement-ws/site' to 'gh-pages' branch and pushing to GitHub. 
INFO    -  Your documentation should shortly be available at: https://adswa.github.io/mpi-datamanagement-ws/ 
```
The rendered documentation should now be available at https://adswa.github.io/mpi-datamanagement-ws/

**Comment by adswa** [comment]
Nice, thanks for setting this up @lnnrtwttkhn. Deploying locally works well for me, the GitHub pages are live after I pushed an initial commit. All subsequent changes in ``docs`` should be live after a ``mkdocs gh-deploy`` as far as I can see for now. I hope I'll get to setting up CI tomorrow. :)

**Comment by lnnrtwttkhn** [comment]
cool, thanks @adswa!

>  the GitHub pages are live after I pushed an initial commit

interesting, I though this would happen automatically with `mkdocs gh-deploy`?

should we keep the documentation on the `docs` branch for now and only merge once we have more content? not sure what's best practice there...

**Comment by adswa** [comment]
Personally, I would be fine either way, but I have a preference for doing everything on master (solely because I might forget that a ``docs`` branch exist :wink: ). I won't be getting to it today anymore unfortunately, but leaving this here as a note to my tomorrow-self: https://github.com/sivaprasadreddy/cheatsheets/blob/master/.github/workflows/main.yml

**Comment by adswa** [comment]
I've played around with everything a bit, and I like it. I gave the ``docs`` versus ``master`` branch another thought and decided that there is no need to make things overly complex by maintaining separate branches for what is hosted via ``mkdocs`` and whatever else may live in this repository on a ``master`` branch. My suggestion is to make ``master`` the default branch to PR all changes against (or simply push into) and I have merged ``docs`` into ``master`` for this. As there is nothing that can break with markdown, I suggest the following workflow for simplicity: Changes that are either deemed controversial, in need for discussion, or would remove meaningful existing contents can be PR'ed from a separate branch to make it easy to request feedback and view change sets. Changes that are small, uncontroversial, discussed on other channels beforehand, etc., can be pushed straight into master.

I have set up a GitHub action that deploys to the GitHub page automatically upon a push to `master` - this saves us a local ``mkdcos gh-deploy``. We could extend this action to other branches or PRs against `master`, if we want to. As there is nothing that can make a markdown page break, continuous integration is not necessary, I suppose.

In general, I like the site that is created by ``mkdocs``. Its simple, elegant, and Markdown makes it easy to add content. However, I think Markdown is a bit limited in what it offers for "discussions" or task structuring (e.g., no way to add comments, lists are always unstructured, discussions via commits, pushes, PR's are a bit slow). My suggestion would be to use a [Google Doc](https://docs.google.com/document/d/1Lw4o5Q3OMjXgGOyFBUHP--_xv01KJtsY76eViMJJLhw/edit) for brainstorming. This has a faster pace, richer structure, and more interactivity. I will add the ideas I have added into the docs in the Google doc shortly.
It would be really cool to have mkdocs generate our course contents, though. Once we have settled for a course structure, we can recreate this structure in the mkdocs toctree and have a syllabus-like GitHub page in the end that will be useful during the workshop and can live on afterwards. 

Another thought for later: If we want to, we could migrate this repository (now, or at any later point) into a more official place than my personal GitHub account, or get a custom domain for the published page. :-)
